ceno_zkvm/src/scheme/verifier.rs

```diff
@@ -148,6 +149,7 @@ impl<E: ExtensionField, PCS: PolynomialCommitmentScheme<E>> ZKVMVerifier<E, PCS>
                 .circuit_vks
                 .get(&name)
                 .ok_or(ZKVMError::VKNotFound(name.clone()))?;
+            transcript.append_field_element(&E::BaseField::from(num_instances as u64));
```

Comment:
In this loop there will be only non-empty circuits. To catch all circuits the `self.vk` should be involved.

Reply:
The verifier now is basically doing the same as the prover, using the information included in the proofs, mimicking the same updates in the corresponding transcripts.